pkg/services/object/server.go

```diff
@@ -523,10 +546,33 @@ type getRangeStreamerV2 struct {
 
 func (s *getRangeStreamerV2) Send(resp *v2object.GetRangeResponse) error {
 	r := resp.ToGRPCMessage().(*protoobject.GetRangeResponse)
-	if err := s.srv.aclChecker.CheckEACL(r, s.reqInfo); err != nil {
-		return eACLErr(s.reqInfo, err)
+	v, ok := r.GetBody().GetRangePart().(*protoobject.GetRangeResponse_Body_Chunk)
+	if !ok {
+		if err := s.srv.aclChecker.CheckEACL(r, s.reqInfo); err != nil {
+			return eACLErr(s.reqInfo, err)
+		}
+		return s.srv.sendRangeResponse(s.ObjectService_GetRangeServer, r)
+	}
+	for buf := bytes.NewBuffer(v.GetChunk()); buf.Len() > 0; {
+		newResp := &protoobject.GetRangeResponse{
+			Body: &protoobject.GetRangeResponse_Body{
+				RangePart: &protoobject.GetRangeResponse_Body_Chunk{
+					Chunk: buf.Next(maxRespDataChunkSize),
+				},
+			},
+			MetaHeader:   proto.Clone(r.GetMetaHeader()).(*protosession.ResponseMetaHeader), // TODO: can go w/o cloning?
+			VerifyHeader: proto.Clone(r.GetVerifyHeader()).(*protosession.ResponseVerificationHeader),
+		}
+		// TODO: do not check response multiple times
```

Comment:
issue? continue work in this PR? find the answers?

Reply:
https://github.com/nspcc-dev/neofs-node/issues/3095
https://github.com/nspcc-dev/neofs-node/issues/3094